Source/WebCore/html/ImageBitmap.cpp

```diff
@@ -782,19 +788,22 @@ class PendingImageBitmap final : public ActiveDOMObject, public FileReaderLoader
             delete this;
         });
 
+        auto completionHandler = std::exchange(m_completionHandler, std::nullopt);
+        ASSERT(completionHandler);
+
         if (!m_arrayBufferToProcess) {
-            m_completionHandler(Exception { InvalidStateError, "An error occured reading the Blob argument to createImageBitmap"_s });
+            completionHandler.value()(Exception { InvalidStateError, "An error occured reading the Blob argument to createImageBitmap"_s });
             return;
         }
 
-        ImageBitmap::createFromBuffer(*scriptExecutionContext(), m_arrayBufferToProcess.releaseNonNull(), m_blob->type(), m_blob->size(), m_blobLoader.url(), WTFMove(m_options), WTFMove(m_rect), WTFMove(m_completionHandler));
+        ImageBitmap::createFromBuffer(*scriptExecutionContext(), m_arrayBufferToProcess.releaseNonNull(), m_blob->type(), m_blob->size(), m_blobLoader.url(), WTFMove(m_options), WTFMove(m_rect), WTFMove(completionHandler.value()));
     }
 
     FileReaderLoader m_blobLoader;
     RefPtr<Blob> m_blob;
     ImageBitmapOptions m_options;
     std::optional<IntRect> m_rect;
-    ImageBitmap::ImageBitmapCompletionHandler m_completionHandler;
+    std::optional<ImageBitmap::ImageBitmapCompletionHandler> m_completionHandler;
```

Comment:
We don't need optional and should't use it. CompletionHandler already has a null state and can be null checked.